app/models/metrics/submit_iteration_metric.rb

```diff
@@ -1,5 +1,7 @@
 class Metrics::SubmitIterationMetric < Metric
   params :iteration
 
+  delegate :exercise, to: :iteration
```

Comment:
You'll also need to do this on various other metric classes. I only did this one as an example :)